File: api/src/main/java/com/hirrao/javaee/controller/BloodPressureController.java
package com.hirrao.javaee.controller;

import com.hirrao.javaee.entity.Result;
import com.hirrao.javaee.entity.User;
import com.hirrao.javaee.service.BloodPressureService;
import com.hirrao.javaee.utils.SnowFlake;
import com.hirrao.javaee.utils.StringUtil;
import com.hirrao.javaee.utils.ThreadLocalUtil;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDate;
import java.util.Map;

import static com.hirrao.javaee.utils.BloodPressureUtil.generateClassification;
import static com.hirrao.javaee.utils.BloodPressureUtil.generateRiskLevel;

@RestController
@RequestMapping("/bp")
public class BloodPressureController {
    private final Logger logger = LoggerFactory.getLogger(getClass());
    private final SnowFlake snowFlake = new SnowFlake(1, 2);
    private final BloodPressureService bloodPressureService;

    public BloodPressureController(BloodPressureService bloodPressureService) {
        this.bloodPressureService = bloodPressureService;
    }

    @PostMapping("/record/uid")
    public Result getUid(@RequestBody Map<String, String> map) {
        logger.debug("/record/uid接受请求{}", map);
        User user = ThreadLocalUtil.get();
        LocalDate date = LocalDate.parse(map.get("date"));
        var result = bloodPressureService.getBloodPressureList(user.getUid(), date);
        return Result.success(result);
    }

    @PostMapping("/record/table")
    public Result getTable(@RequestBody Map<String, String> map) {
        logger.debug("/record/table接受请求{}", map);
        User user = ThreadLocalUtil.get();
        LocalDate date = LocalDate.parse(map.get("date"));
        var result = bloodPressureService.getBloodPressureTable(date, user.getUid());
        return Result.success(result);
    }

    @PostMapping("/record/riskLevel")
    public Result getRiskLevel(@RequestBody Map<String, String> map) {
        logger.debug("/record/riskLevel接受请求{}", map);
        User user = ThreadLocalUtil.get();
        LocalDate date = LocalDate.parse(map.get("date"));
        var result = bloodPressureService.getRiskLevel(user.getUid(), date);
        return Result.success(result);
    }

    @PostMapping("/record/insert")
    public Result insert(@RequestBody Map<String, String> map) {
        logger.debug("/record/insert接受请求{}", map);
        User user = ThreadLocalUtil.get();
        var userId = user.getUid();
        var recordTime = map.get("date");
        var sbp = map.get("sbp");
        var dbp = map.get("dbp");
        if (StringUtil.isEmpty(recordTime) || StringUtil.isEmpty(sbp) || StringUtil.isEmpty(dbp)) {
            return Result.error(101, "参数错误");
        }

        //将string处理为数值进行处理
        float sbpValue = Float.parseFloat(sbp);
        float dbpValue = Float.parseFloat(dbp);

        // 生成 classification 和 riskLevel
        String classification = generateClassification(sbpValue, dbpValue);
        String riskLevel = generateRiskLevel(sbpValue, dbpValue);

        bloodPressureService.insertBloodPressure(snowFlake.nextId(), userId, sbpValue, dbpValue, recordTime, classification, riskLevel);
        return Result.success();
    }
}

File: api/src/main/kotlin/com/hirrao/javaee/controller/BloodPressureController.kt
package com.hirrao.javaee.controller

import com.hirrao.javaee.entity.Result
import com.hirrao.javaee.service.BloodPressureService
import com.hirrao.javaee.utils.BloodPressureUtil.generateClassification
import com.hirrao.javaee.utils.BloodPressureUtil.generateRiskLevel
import com.hirrao.javaee.utils.Result.error
import com.hirrao.javaee.utils.Result.success
import com.hirrao.javaee.utils.SnowFlake
import com.hirrao.javaee.utils.ThreadLocalUtil.get
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate

@RestController
@RequestMapping("/bp")
class BloodPressureController(private val bloodPressureService: BloodPressureService) {
    private val logger: Logger = LoggerFactory.getLogger(javaClass)
    private val snowFlake = SnowFlake(1, 2)

    data class DataInsert(val date: String, val sbp: String, val dbp: String)

    @PostMapping("/record/uid")
    fun getUid(@RequestBody dateRow: String): Result {
        logger.debug("/record/uid接受请求{}", dateRow)
        val user = get()
        val date = dateRow.let { LocalDate.parse(it) }
        val result = bloodPressureService.getBloodPressureList(user.uid, date)
        return success(result)
    }

    @PostMapping("/record/table")
    fun getTable(@RequestBody dateRow: String): Result {
        logger.debug("/record/table接受请求{}", dateRow)
        val user = get()
        val date = dateRow.let { LocalDate.parse(it) }
        val result = bloodPressureService.getBloodPressureTable(date, user.uid)
        return success(result)
    }

    @PostMapping("/record/riskLevel")
    fun getRiskLevel(@RequestBody dateRow: String): Result {
        logger.debug("/record/riskLevel接受请求{}", dateRow)
        val user = get()
        val date = dateRow.let { LocalDate.parse(it) }
        val result = bloodPressureService.getRiskLevel(user.uid, date)
        return success(result)
    }

    @PostMapping("/record/insert")
    fun insert(@RequestBody dataInsert: DataInsert): Result {
        logger.debug("/record/insert接受请求{}", dataInsert)
        val (recordTime, sbp, dbp) = dataInsert
        val user = get()
        val userId = user.uid
        if (recordTime.isEmpty() || sbp.isEmpty() || dbp.isEmpty()) {
            return error(101, "参数错误")
        }

        //将string处理为数值进行处理
        val sbpValue = sbp.toFloat()
        val dbpValue = dbp.toFloat()

        // 生成 classification 和 riskLevel
        val classification = generateClassification(sbpValue, dbpValue)
        val riskLevel = generateRiskLevel(sbpValue, dbpValue)

        bloodPressureService.insertBloodPressure(
            snowFlake.nextId(), userId, sbpValue, dbpValue, recordTime, classification, riskLevel
        )
        return success()
    }
}
